.github/workflows/ci.yml

```
@@ -4,33 +4,35 @@ concurrency:
   group: CI-${{ github.ref }}
   cancel-in-progress: true
 env:
-  PYTHON_VERSION: 3.9
+  PYTHON_VERSION: 3.8  # Works with Pants 1.30.x and Pants 2.x.
 jobs:
   org-check:
     name: Check GitHub Organization
-    if: ${{ github.repository_owner == 'pantsbuild' }}
+    #if: ${{ github.repository_owner == 'pantsbuild' }}
     runs-on: ubuntu-20.04
     steps:
       - name: Noop
         run: "true"
   checks:
-    name:  Lint and check formatting
+    name: Lint and Typecheck
     needs: org-check
     runs-on: ubuntu-20.04
     steps:
       - name: Checkout
         uses: actions/checkout@v3
+      - name: Setup Pants
+        uses: pantsbuild/actions/init-pants@fe0b9d486a7de2f3cbefa62d6bd5e7eefdccee4a
+        with:
+          gha-cache-key: v0-
+          named-caches-hash: ${{ hashFiles('tests/requirements.lock') }}
       - name: Setup Python
         uses: actions/setup-python@v4
         with:
           python-version: ${{ env.PYTHON_VERSION }}
-      - name: Shellcheck
-        run: shellcheck pantsup.sh
-      - name: Run Checks
-        uses: pantsbuild/actions/run-tox@v2
-        with:
-          tox-env: format-check,lint,typecheck
-  unit-tests:
+      - name: Lint and Typecheck
+        run: |
+          pants lint check ::
```

Nit: I find it more useful to get semantic errors first.
```suggestion
          pants check lint ::
```